docs/haskell-use-cases.rst

```diff
@@ -308,20 +308,195 @@ However, you seldom want to use them directly. Cabal packages
 typically have many dependencies, which themselves have dependencies
 and so on. It is tedious to describe all of these dependencies to
 Bazel by hand. You can use the `stack_snapshot`_ workspace rule
-to download the source of all necessary dependencies from Hackage,
-and extract a dependency graph from a Stackage_ snapshot.
+as described below to download the source of all necessary dependencies from
+Hackage, and extract a dependency graph from a Stackage_ snapshot.
 
 These rules are meant only to interoperate with third-party code. For
 code under your direct control, prefer using one of the core Haskell
 rules, which have more features, are more efficient and more
 customizable.
 
+Importing a Stackage snapshot
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `stack_snapshot`_ workspace rule interfaces with the Stack tool to resolve
+package versions and dependencies based on a given Stackage snapshot. It also
+downloads the packages sources and generates Bazel build definitions for the
+individual Cabal packages.
+
+This is how you import the Stackage LTS 14.0 snapshot ::
+
+  stack_snapshot(
+      name = "stackage",
+      snapshot = "lts-14.0",
+      packages = [
+          "base",
+          "optparse-applicative",
+      ],
+  )
+
+This will generate the labels ``@stackage//:base``, and
+``@stackage//:optparse-applicative``, which you can use in the ``deps``
+attribute of your Haskell targets. Note that ``base`` is a core package and its
+version is determined by the GHC toolchain and not the Stackage snapshot.
+
+Use the ``local_snapshot`` attribute to refer to a `custom Stack snapshot`_.
+
+Pinning
+^^^^^^^
+
+The ``stack_snapshot`` rule invokes ``stack`` for version and dependency
+resolution. By default this will happen on every fetch of the repository. You
```

Comment:
Calling `stack` doesn't sound too onerous in of itself, and it's unclear how often or when "fetching" would happen. Perhaps we can tighten this point by making it more clear what the advantages of pinning are to the user? Like, pinning means that users don't even need Stack installed on their system first, right? Also, running Stack means requiring arbitrary network access and that's what makes it slow, right? Also, "repository" here is in the sense of "external repository", but that won't necessarily be all that obvious except for Bazel cognoscenti. Could say "external repository" explicitly, and make it a link to the Bazel doc.

Reply:
I've expanded this to more clearly explain the issues and how pinning resolves them.

Reply:
> Like, pinning means that users don't even need Stack installed on their system first, right?

True, though this one is perhaps a bit less visible to users since rules_haskell will fetch a version of stack if it doesn't find one installed already.